### DISCUSSION TITLE
Integrating DRM into Ant Media Server

### DISCUSSION THREAD

**Discussion by Jason239** [ant-media/Ant-Media-Server#6111] [Plugins/Custom Extensions]
I'm going to use the community version of Ant Media Server and trying to apply custom DRM(Widevine, FPS, PlayReady) to DASH/HLS streams.
Can you tell me how to do it?

**Comment by yashtandon113** [comment]
HI @Jason239 
Currently, there is no such feature available on AMS to use the DRM. However, it is on our roadmap for this year, and we will begin working on it based on user demand.

In order to make your streams secure for now, you can try the Enterprise Edition of Ant Media Server, where you can use token security methods. For more details, check [here](https://antmedia.io/docs/category/stream-security/).

To request your AMS Enterprise Edition free trial, check https://antmedia.io/freetrial/

I hope it helps.

**Comment by Jason239** [comment]
Thanks for reply @yashtandon113 
Okay. So that means there's no way to encrypt the stream through a plugin or anything, right?

**Reply by yashtandon113** [reply]
@Jason239 
Yes, right now, you cannot encrypt the stream using the DRM on AMS.

But as I suggested, you can use the Enterprise Edition of AMS, in which you can use token mechanisms to encrypt your streams.

**Reply by Jason239** [reply]
I got it, thanks! :)

**Comment by Jason239** [comment]
One more thing, is [this EZDRM issue](https://github.com/ant-media/Ant-Media-Server/issues/4724) the roadmap item for this year that you were talking about? Or are you planning to develop an interface that allows to enter DRM information(such as encryption key, DRM PSSH, etc) manually rather than through a specific DRM service?
Or maybe I, or someone else, can implement it themselves?

**Reply by Jason239** [reply]
I'm also curious about roadmap plans for this year. What about this inquiry? @yashtandon113 
Is there anything specific?

**Reply by yashtandon113** [reply]
@Jason239 
Yes, we created the issue to check the feasibility with the EZDRM but we have not decided yet when and which solution we are going to implement.

We will notify the community when there is an update regarding the DRM solution.

**Comment by bijancot** [comment]
Hi @yashtandon113 and teams! thank you for bringing this greate opensource apps. So regarding the original discussion, is implementing drm already supported? or it will take more time to implement? 

Thank you!

**Reply by filip-ams** [reply]
Hi @bijancot,

Thank you for your kind words and for being an active part of our community!

Regarding your question about DRM implementation: this feature is on our roadmap and, when implemented, will most likely be provided as a plugin, similar to how our Low Latency HLS is offered.We typically prioritize feature development based on demand and alignment with our development plans. If you could share more details about your use case or specific requirements, it would help us better understand the demand and potentially expedite its implementation.

Thank you again for your support and valuable feedback!

Best regards,
Filip
